homeassistant/components/volvo/coordinator.py

```diff
@@ -0,0 +1,196 @@
+            else f"{vehicle.description.model} {vehicle.fuel_type} ({vehicle.model_year})"
+        )
+
+        self.device = DeviceInfo(
+            identifiers={(DOMAIN, vehicle.vin)},
+            manufacturer=MANUFACTURER,
+            model=model,
+            name=f"{MANUFACTURER} {vehicle.description.model}",
+            serial_number=vehicle.vin,
+        )
+
+        self._refresh_conditions = {
+            "command_accessibility": (self.api.async_get_command_accessibility, True),
+            "diagnostics": (self.api.async_get_diagnostics, True),
+            "fuel": (
+                self.api.async_get_fuel_status,
+                self.vehicle.has_combustion_engine(),
+            ),
+            "odometer": (self.api.async_get_odometer, True),
+            "recharge_status": (
+                self.api.async_get_recharge_status,
+                self.vehicle.has_battery_engine(),
+            ),
+            "statistics": (self.api.async_get_statistics, True),
+        }
+
+    async def _async_update_data(self) -> CoordinatorData:
+        """Fetch data from API."""
+
+        api_calls = self._get_api_calls()
+        data: CoordinatorData = {}
+        valid = 0
+        exception: Exception | None = None
+
+        results = await asyncio.gather(
+            *(call() for call in api_calls), return_exceptions=True
+        )
+
+        for result in results:
+            if isinstance(result, VolvoAuthException):
+                # If one result is a VolvoAuthException, then probably all requests
+                # will fail. In this case we can cancel everything to
+                # reauthenticate.
+                #
+                # Raising ConfigEntryAuthFailed will cancel future updates
+                # and start a config flow with SOURCE_REAUTH (async_step_reauth)
+                _LOGGER.exception(
+                    "%s - Authentication failed. %s",
+                    self.config_entry.entry_id,
+                    result.message,
+                )
+                raise ConfigEntryAuthFailed(
+                    f"Authentication failed. {result.message}"
+                ) from result
+
+            if isinstance(result, VolvoApiException):
+                # Maybe it's just one call that fails. Log the error and
+                # continue processing the other calls.
+                _LOGGER.debug(
+                    "%s - Error during data update: %s",
+                    self.config_entry.entry_id,
+                    result.message,
+                )
+                exception = exception or result
+                continue
+
+            if isinstance(result, Exception):
+                # Something bad happened, raise immediately.
+                raise UpdateFailed(
+                    translation_domain=DOMAIN,
+                    translation_key="update_failed",
+                ) from result
+
+            data |= cast(CoordinatorData, result)
+            valid += 1
```

Comment:
Since we don't care how many calls failed or not, perhaps just use an error variable set to true or false instead of a counter we won't really use for anything